app/lua/lnodemcu.h

```diff
@@ -45,9 +45,11 @@
   static ROTable_entry LOCK_IN_SECTION(s) rt ## _entries[] = {
 #define LROT_END(rt,mt,f)    {NULL, LRO_NILVAL} }; \
   const ROTable rt ## _ROTable = { \
-    (GCObject *)1, LUA_TROTABLE, LROT_MARKED, \
-    cast(lu_byte, ~(f)), (sizeof(rt ## _entries)/sizeof(ROTable_entry)) - 1, \
-    cast(Table *, mt), cast(ROTable_entry *, rt ## _entries) };
+    .next = (GCObject *)1, .tt = LUA_TROTABLE, .marked = LROT_MARKED, \
```

Comment:
This is quite a change to the ROTable layout.  The current code uses exploit the fact that a common subset of `Table` and `ROTable` fields are effectively a superset union, so the code doesn't need messy conditional selectors.  This change breaks this assumption.   I need to think through the implications.